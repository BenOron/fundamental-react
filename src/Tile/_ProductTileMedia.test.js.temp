import { mount } from 'enzyme';
import ProductTile from './ProductTile';
import React from 'react';
import renderer from 'react-test-renderer';

describe('<ProductTile.Media />', () => {
    const productTileMedia = (
        <ProductTile.Media image='https://techne.yaas.io/images/product-thumbnail-wide.png' />
    );

    test('create ProductTile.Media component', () => {
        let component = renderer.create(productTileMedia);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the ProductTileMedia component', () => {
            const element = mount(<ProductTile.Media data-sample='Sample' image='https://techne.yaas.io/images/product-thumbnail-wide.png' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
});
