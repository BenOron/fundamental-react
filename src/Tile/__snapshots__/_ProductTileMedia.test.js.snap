// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductTile.Media /> create ProductTile.Media component 1`] = `
<div
  className="fd-product-tile__media"
  style={
    Object {
      "backgroundImage": "url(https://techne.yaas.io/images/product-thumbnail-wide.png)",
    }
  }
/>
`;
